sincere-singularities/src/sincere_singularities/modules/order_queue.py

```diff
@@ -0,0 +1,222 @@
+    """
+    try:
+        return save_states.load_game_state(user_id)["number_of_orders"][restaurant]
+    except (ValueError, KeyError):
+        return 0
+
+
+def add_number_of_orders(user_id: int, restaurant: str) -> None:
+    """
+    Add to the number of orders.
+
+    Args:
+        user_id (int): The user's ID.
+        restaurant (str): The restaurant's name.
+    """
+    try:
+        state = save_states.load_game_state(user_id)
+    except ValueError:
+        state = generate_default_state()
+
+    state["number_of_orders"].setdefault(restaurant, 0)
+    state["number_of_orders"][restaurant] += 1
+    save_states.save_game_state(user_id, state)
+
+
+class OrderQueue:
+    """The class for managing the order queue. Orders can be spawned and deleted from here."""
+
+    def __init__(self, interaction: ApplicationCommandInteraction, webhook: Webhook) -> None:
+        """
+        Initialize the order queue.
+
+        Args:
+            interaction (ApplicationCommandInteraction): The application command interaction.
+            webhook (Webhook): The webhook.
+        """
+        self.interaction = interaction
+        self.user = interaction.user
+        self.orders: dict[str, tuple[Order, WebhookMessage]] = {}
+        self.running = False
+        self.webhook = webhook
+        self.order_generators: defaultdict[str, OrderGenerator] = defaultdict(lambda: OrderGenerator(Difficulty.EASY))
+        self.orders_thread: Thread | None = None
+
+    @classmethod
+    async def new(cls, interaction: ApplicationCommandInteraction) -> Self | None:
+        """
+        Create a new order queue.
+
+        Args:
+            interaction (ApplicationCommandInteraction): The application command interaction.
+
+        Returns:
+            Self | None: The new order queue, or None if a webhook couldn't be created.
+        """
+        if not isinstance(interaction.channel, TextChannel):
+            raise TypeError("interaction.channel should be TextChannel")
+        try:
+            webhook = await interaction.channel.create_webhook(name="Restaurant Rush: Kitchen Chaos - Order Webhook")
+        except CommandInvokeError:
+            await interaction.channel.send(
+                "Can't start Restaurant Rush: Kitchen Chaos: maximum amount of webhooks reached. "
+                "Delete webhooks or try in another channel!"
+            )
+            return None
+
+        return cls(
+            interaction=interaction,
+            webhook=webhook,
+        )
+
+    async def start_orders(self) -> None:
+        """Start the orders queue. Spawns three orders."""
+        self.running = True
+
+        # Creating Orders Thread after initial Message was sent for proper message order.
+        self.orders_thread = await self.interaction.channel.create_thread(
+            name="Orders Thread", type=ChannelType.public_thread, invitable=False
+        )
+        assert self.orders_thread  # MYPY can be stupid at times...
```

Comment:
Creating the thread inside the `.new()` classmethod should avoid some of these type issues that Pyright still errors on.
Does "proper message order" mean that the thread creation message must come after the buttons view message? If so, why?